test/test.dart

```diff
@@ -551,21 +551,29 @@ Future<User> loginWithRetry(App app, Credentials credentials, {int retryCount =
 }
 
 Future<void> waitForCondition(
-  bool Function() condition, {
+  FutureOr<bool> Function() condition, {
   Duration timeout = const Duration(seconds: 1),
   Duration retryDelay = const Duration(milliseconds: 100),
   String? message,
-}) async {
-  await Future.any<void>([
-    Future<void>.delayed(timeout, () => throw TimeoutException('Condition not met within $timeout. Message: ${message != null ? ': $message' : ''}')),
-    Future.doWhile(() async {
-      if (condition()) {
-        return false;
-      }
-      await Future<void>.delayed(retryDelay);
-      return true;
-    })
-  ]);
+}) {
+  return waitForConditionWithResult<bool>(() => condition(), (value) => value == true);
```

Comment:
```suggestion
return waitForConditionWithResult<bool>(() => condition(), (value) => value);
```

Reply:
Fixed in https://github.com/realm/realm-dart/pull/975.